website/docs/artist_hosts_maya_xgen.md

```diff
@@ -4,26 +4,92 @@ title: Xgen for Maya
 sidebar_label: Xgen
 ---
 
-## Working with Xgen in OpenPype
+## Setup
 
-OpenPype support publishing and loading of Xgen interactive grooms. You can publish 
-them as mayaAscii files with scalps that can be loaded into another maya scene, or as
-alembic caches. 
+### Settings
 
-### Publishing Xgen Grooms
+Go to project settings > Maya > enable "Open Workfile Post Initialization";
 
-To prepare xgen for publishing just select all the descriptions that should be published together and the create Xgen Subset in the scene using - **OpenPype menu** → **Create**... and select **Xgen Interactive**. Leave Use selection checked.
+`project_settings/maya/open_workfile_post_initialization`
 
-For actual publishing of your groom to go **OpenPype → Publish** and then press ▶ to publish. This will export `.ma` file containing your grooms with any geometries they are attached to and also a baked cache in `.abc` format 
+This is due to two errors occurring when opening workfile contaiing referenced xgen nodes on launch of Maya;
 
+- ``Critical``: Duplicate collection errors on launching workfile. This is because Maya first imports Xgen when referencing in external Maya files, then imports Xgen again when the reference edits are applied.
+```
+Importing XGen Collections...
+# Error: XGen:  Failed to find description ball_xgenMain_01_:parent in collection ball_xgenMain_01_:collection. Abort applying delta: P:/PROJECTS/OP01_CG_demo/shots/sh040/work/Lighting/cg_sh040_Lighting_v001__ball_xgenMain_01___collection.xgen  #
+# Error: XGen:  Tried to import a duplicate collection, ball_xgenMain_02_:collection, from file P:/PROJECTS/OP01_CG_demo/shots/sh040/work/Lighting/cg_sh040_Lighting_v001__ball_xgenMain_02___collection.xgen. Aborting import.  #
+```
+- ``Non-critical``: Errors on opening workfile and failed opening of published xgen. This is because Maya imports Xgen when referencing in external Maya files but the reference edits that ensure the location of the Xgen files are correct, has not been applied yet.
+```
+Importing XGen Collections...
+# Error: XGen:  Failed to open file: P:/PROJECTS/OP01_CG_demo/shots/sh040/work/Lighting/cg_ball_xgenMain_v035__ball_rigMain_01___collection.xgen  #
+# Error: XGen:  Failed to import collection from file P:/PROJECTS/OP01_CG_demo/shots/sh040/work/Lighting/cg_ball_xgenMain_v035__ball_rigMain_01___collection.xgen  #
+```
 
-:::tip adding more descriptions
-You can add multiple xgen description into the subset you are about to publish, simply by
-adding them to the maya set that was created for you. Please make sure that only xgen description nodes are present inside of the set and not the scalp geometry. 
-:::
+### Workfile Incremental Save
 
-### Loading Xgen
+When you increment the Maya workfile to a new version, all `.xgen` and `.xgd` files referenced by the Xgen collection in the workspace is incremented as well.
```

Comment:
```suggestion
When you increment the Maya workfile to a new version, all `.xgen` and `.xgd` files referenced by the Xgen collection in the workspace are incremented as well.
```

Reply:
Reading this sentence it leaves me wondering what does this means for my assets? Does it need to recache? Why does it need to increment along automatically?

Reply:
Yeah, I'm not sure this feature is actually needed in the end, or it needs to at least be configurable. There is no reason other than keeping the work files and area tidy.
The thinking was when publishing and the workfile gets incremented, we are still using the old versions xgen files so it would be hard (impossible?) to go back to an older workfile straight from the Workfile app.

The feature is actually also lacking because its only incrementing the `.xgen` and `.xgd` files, and not the side xgen folder structure containing `.ptx` and `.xuv`.

Question is whether this feature is worth finishing or abandon? @antirotor @mkolar @BigRoy